app/mailers/user_notifications.rb

```diff
@@ -349,13 +349,29 @@ def user_locale(user)
   end
 
   def email_post_markdown(post, add_posted_by = false)
-    result = +"#{post.raw}\n\n"
+    result = +"#{post.with_secure_media? ? strip_secure_urls(post.raw) : post.raw}\n\n"
     if add_posted_by
       result << "#{I18n.t('user_notifications.posted_by', username: post.username, post_date: post.created_at.strftime("%m/%d/%Y"))}\n\n"
     end
     result
   end
 
+  def strip_secure_urls(raw)
+    urls = Set.new
+    raw.scan(URI.regexp(%w{http https})) { urls << $& }
+
+    urls.each do |url|
+      begin
+        if (url.start_with?(Discourse.store.s3_upload_host) && FileHelper.is_supported_media?(url))
+          raw = raw.sub(url, "<p class='secure-media-notice'>#{I18n.t("emails.secure_media_placeholder")}</p>")
+        end
+      rescue URI::InvalidURIError
```

Comment:
What could be throwing this exception?

Reply:
Any bad URL, for example `http:error]`. See also https://github.com/discourse/discourse/blob/master/spec/lib/search_spec.rb#L46

Reply:
I meant both `start_with?` and `FileHelper.is_supported_media?` are working on a `string` and, as far as I know, are not trying to convert the `string` into an `URI` object.

Reply:
D'oh... 🤦‍♂ Cleaned up now.